### PR TITLE
Refactor scos-sensor to support plugins for different actions and signal analyzers

### PR DESCRIPTION
**Pull request by jhazentia**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,138 +42,303 @@ monitoring.
...
+  no start time is given, the first task is scheduled as soon as possible. If no stop
+  time is given, tasks continue to be scheduled until the schedule entry is manually
+  deactivated. Leaving the interval undefined results in a "one-shot" entry, where the
+  scheduler deactivates the entry after a single task is scheduled. One-shot entries
+  can be used with a future start time. If two tasks are scheduled to run at the same
+  time, they will be run in order of *priority*. If two tasks are scheduled to run at
+  the same time and have the same *priority*, execution order is
+  implementation-dependent (undefined).
+
+- *signal*: Django notifications library used to allow scos-sensor to receive
```
Comment (dboulware) [inline README.md:136]: action completion is just one example right? Isn't a broader framework for event driven programming/architecture?

File: README.md [hunk 2]
```diff
@@ -42,138 +42,303 @@ monitoring.
...
+- A website and JSON RESTful API using [Django REST framework] is served over HTTPS via
+  [NGINX], a high-performance web server. These provide easy administration over the
+  sensor.
+
+![SCOS Sensor Architecture Diagram](/docs/img/architecture_diagram.png?raw=true)
 
+A functioning scos-sensor utilizes software from at least three different GitHub
+repositories. As shown below, the scos-sensor repository integrates everything together
+as a functioning scos-sensor and provides the code for the user interface, scheduling,
+and the storage and retrieval of schedules and acquisitions. The scos-actions
```
Comment (dboulware) [inline README.md:177]: add link to scos-actions

File: README.md [hunk 3]
```diff
@@ -42,138 +42,303 @@ monitoring.
...
+repository provides the core actions API, defines the radio interface that provides an
+abstraction for all signal analyzers, and provides basic actions. Finally, using a real
+radio within scos-sensor requires a third `scos-<signal analyzer>` repository that
+provides the signal analyzer specific implementation of the radio interface where
+`<signal analyzer>` is replaced with the name of the signal analyzer, e.g. a USRP
+scos-sensor utilizes the scos-usrp repository. The signal analyzer specific
```
Comment (dboulware) [inline README.md:183]: add link to scos-usrp

File: docker-compose.yml [hunk 4]
```diff
@@ -26,35 +26,39 @@ services:
...
+      - DEBUG
       - DOCKER_TAG
+      - DOMAINS
+      - FQDN
       - GIT_BRANCH
+      - GUNICORN_LOG_LEVEL
       - IN_DOCKER=1
+      - IPS
+      - MAX_DISK_USAGE
+      - MAX_TASK_RESULTS
```
Comment (dboulware) [inline docker-compose.yml:49]: Shouldn't MAX_TASK_RESULTS be removed?

Comment (jhazentia) [reply docker-compose.yml:49]: Good catch, I missed this when merging master into this branch

File: src/actions/fft.py [hunk 5]
```diff
@@ -1,6 +1,87 @@
+from enum import Enum
```
Comment (dboulware) [inline src/actions/fft.py:1]: Shouldn't this have moved to scos_actions?

File: README.md [hunk 6]
```diff
@@ -290,8 +290,8 @@ settings in the environment file:
 - IPS: A space separated list of IP addresses. Used to generate [ALLOWED_HOSTS](
   https://docs.djangoproject.com/en/3.0/ref/settings/#allowed-hosts).
 - FQDN: The server’s fully qualified domain name.
-- MAX_TASK_RESULTS: The maximum number of task results to keep before overwriting old
-  results. Defaults to 100,000.
+- MAX_DISK_USAGE: The maximum disk usage percentage allowed before overwriting old
```
Comment (dboulware) [inline README.md:293]: Should we add a note that there will be some difference with the usage reported in Linux?

**Review by dboulware — APPROVED** [review]
Nice job.

**Pull request merged**